augur/export_v2.py

```diff
@@ -211,8 +212,32 @@ def _get_title(key):
         return key
 
     def _add_color_scale(coloring):
+        ## consider various sources to find any user-provided scale information
         key = coloring["key"]
-        if key.lower() in provided_colors:
+        scale_type = coloring["type"]
+        if scale_type is "continuous":
+            ## continuous scale information can only come from an auspice config JSON
+            if config.get(key, {}).get("scale"):
+                # enforce numeric values (we can't use the schema for this)
+                provided_scale = [s for s in config[key]['scale'] if isinstance(s[0], numbers.Number)]
+                if len(provided_scale)<2:
+                    warn(f"The scale provided for {key} had fewer than 2 (valid numeric) entries. Skipping.")
+                    return coloring
+                coloring["scale"] = provided_scale
+                return coloring
+        elif config.get(key, {}).get("scale"):
+            # we preferentially use a scale explicitly provided in the auspice config JSON
+            values_in_tree = get_values_across_nodes(node_attrs, key)
```

Comment:
Really interested in talking through this -- in my original implementation (which was years ago) I really liked using closures like this to both avoid having to pass small functions lots of variables, especially when recursion was involved, and to indicate clearly that this function was only used in this one place (in fact, I don't think it can be called from anywhere else, which is problematic for testing!).

Reply:
Resolving as this new code follows the style of the existing code, and bigger changes are out-of-scope for this PR